logic.typ

```diff
@@ -159,6 +159,50 @@
   }
 }
 
+#let fill-remaining-height(
+  margin: 0%,
+  before-label: "before-fill",
+  ..box-kwargs,
+  h-align: left,
+  content,
+) = {
+  locate(loc => {
```

Comment:
Couldn't we introduce the label at the beginning of the function ourselves?

Reply:
`label` must be attached to a content block, i.e., if you put the label at the beginning of the function, `typst` will raise an error regarding the orphaned label.

Reply:
If that does not work, maybe we could provide
```typ
#let fit-marker = {
  #show raw.where(lang: "fit-remaining-marker"): []
  #raw("abc", lang: "fit-remaining-marker") <fit-remaining-label>
}
```
to hide that implementation detail a bit.

Reply:
> `label` must be attached to a content block

Okay yeah we could use that `raw` trick for that, see other comment.

Reply:
Great idea. Turns out, it can be simplified even further by applying the `show` rule to the label itself -- no need for `raw` :)

Reply:
Ok, so there is one downside to this approach -- how much space the function identifies. Here is the layout when <fill-remaining-marker> is user-specified:

![image](https://github.com/andreasKroepelin/polylux/assets/23620506/130bdcc2-74df-43f8-ae62-37c0a5bd15f4)


The same code, but with the label programmatically placed and hidden:
![image](https://github.com/andreasKroepelin/polylux/assets/23620506/a5c56a93-7630-4bbf-9d2a-3df267d9b24d)


Any thoughts on a solution? Is it robust enough to always add `1em` to the detected height, assuming this is the height of the hidden label?

Reply:
I think it is too hacky and unreliable (and possibly wrong?) to assume that `[]` has a height of 1em. Does it help to change the show rule to
```typ
#show ...: none
```
?

Reply:
`show ...: none` has the same effect as `[]`.

You are correct -- it appears the discrepancies are more to do with the slide margins than deficiencies with size calculations. I've made a project to confirm this:
https://typst.app/project/rP3GCbwICL7rZhPvTML3K-

It is clear that no matter what size text is set, `fill-remaining-height` correctly identifies the space available **without** requiring a fudge factor

![image](https://github.com/andreasKroepelin/polylux/assets/23620506/4f106f4e-fe98-4a2b-8caf-2acc5a331bdb)

Therefore, based on your valid recommendation, I've reverted this adjustment 👍